tools/gfx/d3d12/render-d3d12.cpp

```diff
@@ -4871,7 +4930,24 @@ Result D3D12Device::createBufferResource(const IBufferResource::Desc& descIn, co
     bufferDesc.Flags |= _calcResourceFlags(srcDesc.allowedStates);
 
     const D3D12_RESOURCE_STATES initialState = buffer->m_defaultState;
-    SLANG_RETURN_ON_FAIL(createBuffer(bufferDesc, initData, srcDesc.sizeInBytes, buffer->m_uploadResource, initialState, buffer->m_resource));
+    SLANG_RETURN_ON_FAIL(createBuffer(bufferDesc, initData, srcDesc.sizeInBytes, buffer->m_uploadResource, initialState, buffer->m_resource, descIn.isShared));
+
+    returnComPtr(outResource, buffer);
+    return SLANG_OK;
+}
+
+Result D3D12Device::createBufferFromNativeHandle(IResource::NativeResourceHandle handle, const IBufferResource::Desc& srcDesc, IBufferResource** outResource)
```

Comment:
No need to do this now, but we should plan to implement `createBufferFromSharedHandle` and `createTextureFromSharedHandle` by using `ID3D12Device::ID3D12Device::OpenSharedHandle`.

Reply:
That's the plan after we get buffer sharing from Vulkan to CUDA working. `createTextureFromSharedHandle` also needs to be added for CUDA, but the process seems to be very similar to sharing a buffer (with an extra step added to handle mipmaps) and seemed a little lower priority at the moment.